### ISSUE TITLE
[Tests]: Increase PHPUnit coverage.

### ISSUE THREAD

**Issue by costdev** [aspirepress/aspireupdate#216]
This is a tracking issue for increasing PHPUnit test coverage of the codebase.

Below is an outline of each class, its assignee, its dependencies, and its methods.

---

### Admin_Settings
- Assigned to @costdev
- Depends on #212

<details>
<summary>
<h4>Click to view the methods</h4>
</summary>

- [x] `::__construct()` - See #229 
- [x] `::get_instance()` - See #230 
- [x] `::get_default_settings()` - See #234 
  - This is a `private` method.
  - It can be covered by `::get_setting()` tests.
- [x] `::reset_settings()` - See #231 
- [x] `::delete_all_settings()` - See #232 
- [x] `::admin_notices()` - See #233 
- [x] `::get_setting()` - See #234 
  - This is partially covered by existing tests.
- [x] `::get_settings_from_config_file()` - See #234 
  - This is a `private` method.
  - It can be covered by `::get_setting()` tests.
- [x] `::update_settings()` - See #235 
- [x] `::register_admin_menu()` - See #236 
- [x] `::admin_enqueue_scripts()` - See #237 
- [x] `::the_settings_page()` - See #238 
- [x] `::register_settings()` - See #241
- [x] `::add_settings_field_callback()` - See #251 
  - This method outputs only HTML and has no conditions.
  - This should not be covered by PHPUnit tests. We can test with E2E instead when we get that set up.
- [x] `::sanitize_settings()` - See #239 
</details>

---

### API_Rewrite
- Assigned to @costdev

<details>
<summary>
<h4>Click to view the methods</h4>
</summary>

- [x] `::__construct()` - See #217 
- [x] `::pre_http_request()` - See #218 
</details>

---

### AspireCloud_API
- Not implemented at this time.
- No PHPUnit tests needed.

---

### Branding
- Already covered.

<details>
<summary>
<h4>Click to view the methods</h4>
</summary>

- [x] `::__construct()`
- [x] `::get_instance()`
- [x] `::admin_enqueue_scripts()`
- [x] `::output_admin_notice()`
</details>

---

### Controller
- Assigned to @costdev

<details>
<summary>
<h4>Click to view the methods</h4>
</summary>

- [x] `::__construct()` - See #219 
- [x] `::api_rewrite()`
  - This is a `private` method.
  - Cannot be tested without testing other class' methods which will already be tested.
  - This should not be covered by PHPUnit tests.
- [x] `::clear_log()`
  -  This is an AJAX callback which calls `wp_send_json_success|error()`, which call `die()` internally.
  - It cannot be tested because script execution is terminated.
  - This _cannot_ be covered by PHPUnit tests.
- [x] `::read_log()`
  -  This is an AJAX callback which calls `wp_send_json_success|error()`, which call `die()` internally.
  - It cannot be tested because script execution is terminated.
  - This _cannot_ be covered by PHPUnit tests.
</details>

---

### Debug
- Assigned to @costdev
- Want to handle this one? Drop a comment on this issue!

<details>
<summary>
<h4>Click to view the methods</h4>
</summary>

- [x] `::get_file_path()`
  - This is a `private` method.
  - It can be covered by `::read()`, `::clear()` or `::log()` tests.
  - See #245
  - See #246
  - See #247
- [x] `::init_filesystem()` 
  - This is a `private` method.
  - It can be covered by `::read()`, `::clear()` or `::log()` tests.
  - See #245
  - See #246
  - See #247
- [x] `::verify_filesystem()`
  - This is a `private` method.
  - It can be covered by `::read()`, `::clear()` or `::log()` tests.
  - See #245
  - See #246
  - See #247
- [x] `::read()` - See #245 
- [x] `::clear()` - See #246 
- [x] `::log()` - See #247 
- [x] `::format_message()` - See #247 
  - This is a `private` method.
  - It can be covered by `::log()` tests.
- [x] `::log_string()` - See #248 
- [x] `::log_request()` - See #249 
- [x] `::log_response()` - See #250 
</details>

---

### Plugins_Screens
- Assigned to @costdev

<details>
<summary>
<h4>Click to view the methods</h4>
</summary>

- [x] `::__construct()` - See #220 
- [x] `::get_instance()` - See #221 
- [x] `::remove_unused_filter_tabs()` - See #222 
</details>

---

### Themes_Screens
- Assigned to @costdev

<details>
<summary>
<h4>Click to view the methods</h4>
</summary>

- [x] `::__construct()` - See #223 
- [x] `::get_instance()` - See #224 
- [x] `::admin_enqueue_scripts()` - See #225 
- [x] `::redirect_to_theme_install()` - See #226 
</details>

---

### Utilities
- Assigned to @costdev
- Depends on #194

<details>
<summary>
<h4>Click to view the methods</h4>
</summary>

- [x] `::get_site_domain()` - See #228 
  - This method is currently named `::get_top_level_domain()`.
  - It will be replaced by #194.
- [x] `::include_file()` - See #227 
</details>

**Comment by costdev** [comment]
Line and branch coverage of the codebase at the time of writing is now at 100%.

Here's a merged single-site and multisite coverage report showing line coverage. Merging coverage reports doesn't support branch coverage unfortunately.

![image](https://github.com/user-attachments/assets/524759bb-fc2e-4ae3-86be-37e7a5b481d2)

Closing out this issue.
